test/integration_test.py

```diff
@@ -985,16 +986,14 @@ def _get_indexed_bundles(self,
                     version=bundle['bundleVersion']
                 )
                 if config.is_anvil_enabled(catalog):
-                    # Biosamples are used as bundle entities for primary bundles
-                    # and are never present in supplementary bundles.
-                    # We cannot use the `files.is_supplementary` field to
-                    # determine whether a bundle is supplementary or not due to
-                    # false positives in the current snapshots
-                    # (https://github.com/DataBiosphere/azul/issues/5229)
+                    is_supplementary = one(set.union(*(
+                        set(always_iterable(file['is_supplementary']))
```

Comment:
Why are the inner sets needed? Wouldn't `set(chain.from_iterable(…` be a better choice here?